pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -94,13 +94,20 @@ func (p Provisioner) Provision(options *apibkt.BucketOptions) (*bktv1alpha1.Obje
 		return nil, err
 	}
 
-	_, errCode, err := cephObject.SetQuotaUserBucketMax(p.objectContext, p.cephUserName, maxBuckets)
-	if errCode > 0 {
+	_, err = cephObject.SetQuotaUserBucketMax(p.objectContext, p.cephUserName, maxBuckets)
+	if err != nil {
 		p.deleteOBCResourceLogError(p.bucketName)
 		return nil, err
 	}
 	logger.Infof("set user %q bucket max to %d", p.cephUserName, maxBuckets)
 
+	// setting quota limit if it is enabled
+	err = p.setAdditionalSettings(options)
+	if err != nil {
+		p.deleteOBCResource(p.bucketName)
```

Comment:
@thotz here we need to handle the error this will lead Gosec CI to fail.  Gosec error  [code](https://github.com/rook/rook/issues/3765#issuecomment-658382312) we are handling or suppressing